spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java

```diff
@@ -95,36 +98,43 @@ public void watchConfigKeyValues() {
         }
 
         for (ConfigStore configStore : configStores) {
-            if (needRefresh(configStore)) {
+            if (needRefreshConfiguration(configStore) || needRefreshFeatureFlag(configStore)) {
                 break;
             }
         }
     }
 
-    private boolean needRefresh(ConfigStore store) {
+    private boolean needRefreshConfiguration(ConfigStore store) {
         String watchedKeyNames = watchedKeyNames(store, storeContextsMap);
+        return needRefresh(store, CONFIGURATION_STORE, watchedKeyNames);
+    }
+    
+    private boolean needRefreshFeatureFlag(ConfigStore store) {
+        return needRefresh(store, FEATURE_STORE, FEATURE_STORE_WATCH_KEY);
+    }
+    
+    private boolean needRefresh(ConfigStore store, String storeName, String watchedKeyNames) {
         QueryOptions options = new QueryOptions().withKeyNames(watchedKeyNames)
                 .withLabels(store.getLabels()).withFields(QueryField.ETAG).withRange(0, 0);
 
         List<KeyValueItem> keyValueItems = configOperations.getRevisions(store.getName(), options);
+        
         if (keyValueItems.isEmpty()) {
             return false;
         }
 
         String etag = keyValueItems.get(0).getEtag();
-        if (firstTimeMap.get(store.getName()) == null) {
-            storeEtagMap.put(store.getName(), etag);
-            firstTimeMap.put(store.getName(), false);
+        if (firstTimeMap.get(store.getName() + storeName) == null) {
+            storeEtagMap.put(store.getName() + storeName, etag);
+            firstTimeMap.put(store.getName() + storeName, false);
```

Comment:
nit: The `storeName` in the parameter is not actually storeName but a suffix, will it cause any confusion?